src/Microsoft.AspNetCore.Razor.Language/Extensions/DefaultTagHelperTargetExtension.cs

```diff
@@ -637,6 +638,33 @@ private string GetContent(HtmlContentIntermediateNode node)
             return builder.ToString();
         }
 
+        // Internal for testing
+        internal static string GetDeterministicId(CodeRenderingContext context)
+        {
+            // Use the file checksum along with the absolute position in the generated code to create a unique id for each tag helper call site.
+            var checksumBytes = context.SourceDocument.GetChecksum();
+            var absoluteIndexBytes = BitConverter.GetBytes(context.CodeWriter.Location.AbsoluteIndex);
+            if (BitConverter.IsLittleEndian)
+            {
+                // We want to represent int using big endian notation.
+                // Rfc1014#section-3.1
+                Array.Reverse(absoluteIndexBytes);
+            }
+            var combinedBytes = checksumBytes.Concat(absoluteIndexBytes).ToArray();
+
+            // Use MD5 to create a 16 byte hash that can be parsed by Guid.
+            using (var md5 = MD5.Create())
```

Comment:
This will get flagged by all kinds of security reviews, even though this might have nothing to do with security. Can we use some other mechanism to avoid getting flagged at all?

Reply:
The other alternative is to use SHA1 which creates a 20 byte hash and only use the first 16 bytes from it. Which should be fine for our use case. Thoughts?

Reply:
So you're OK with breaking someone's code if they did parse them as guids? 😆

Reply:
Yes, I am. It didn't even look like a GUID (there were no dashes or curly braces). It was just a 16-byte hex string.

Reply:
Oh ok, #YOLO then - I thought we used to use the GUID format, but I am mistaken (first time evar)